### PR TITLE
Allow passing of custom MS for per

### PR DESCRIPTION
**Pull request by darksheik**
I'd like to be able to determine my own interval.  I have a case where I need to enforce 1 request every 10 seconds.

Thanks!

### CODE REVIEW

**Comment by facundoolano** [comment]
Can you add a test for this?

**Comment by darksheik** [comment]
Sure!  I'm honestly more of an Elixir guy and I'm not sure if I'm running the suite correctly...

```
Common Test v1.13 starting (cwd is /Users/don/workspace/throttle)

Eshell V8.2  (abort with ^G)
ct_run -dir /Users/don/workspace/throttle

(ct@MacBook-Pro)1> 
Common Test: Running make in test directories...

CWD set to: "/Users/don/workspace/throttle/ct_run.ct@MacBook-Pro.2018-06-13_13.49.03"

TEST INFO: 1 test(s), 2 suite(s)

Testing workspace.throttle: Starting test (with repeated test cases)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
throttle_distributed_SUITE:init_per_suite failed
Reason: {badmatch,{error,boot_timeout,'node2@127.0.0.1@MacBook-Pro'}}
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Testing workspace.throttle: *** FAILED {throttle_distributed_SUITE,init_per_suite} ***

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
throttle_test_SUITE:init_per_group failed
Reason: {badmatch,{error,{throttle,{"no such file or directory","thr...}
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Testing workspace.throttle: *** FAILED {throttle_test_SUITE,init_per_group} ***

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
throttle_test_SUITE:init_per_group failed
Reason: {badmatch,{error,{throttle,{"no such file or directory","thr...}
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Testing workspace.throttle: *** FAILED {throttle_test_SUITE,init_per_group} ***
Testing workspace.throttle: TEST COMPLETE, 0 ok, 0 failed, 19 skipped of 19 test cases

Updating /Users/don/workspace/throttle/index.html ... done
Updating /Users/don/workspace/throttle/all_runs.html ... done
```

**Comment by facundoolano** [comment]
Did you run them with `make test`? What version of erlang?

**Comment by darksheik** [comment]
@facundoolano Thanks!  That worked.  Test added.

File: src/throttle_time.erl [hunk 1]
```diff
@@ -14,7 +14,9 @@ interval(per_hour) ->
 interval(per_minute) ->
   1000 * 60;
 interval(per_second) ->
-  1000.
+  1000;
+interval(CustomMs) ->
```
Comment (facundoolano) [inline src/throttle_time.erl]: can you add a `when is_integer(CustomMs)` here, so we still break on malformed input?

**Comment by facundoolano** [comment]
Thanks, added one last minor comment.

File: src/throttle_time.erl (hunk 1 above)
Comment (darksheik) [reply src/throttle_time.erl]: Done!

**Pull request merged**